matlab_simulation/05-estimation/fastSLAM/featureSearch_fs.m

```diff
@@ -0,0 +1,36 @@
+%% FUNC to search of features in Field of VIEW
+% INPUTS:(in order of call)
+% -------
+% No of features
+% Map of the Environment
+% Robot's pose at the current time step
+% Maximum Range
+% Maximum FOV
+% OUTPUTS:(in order of call)
+% --------
+% A vector containing all the measured indexs
+% % AUTHOR: BISMAYA SAHOO, EMAIL:bsahoo@uwaterloo.ca
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Feature Search
+    function meas_ind=featureSearch_fs(noFeatures,map,robot_pose,rmax,thmax)
+        meas_ind = [];
+        for i=1:noFeatures
+                if(inview(map(:,i),robot_pose,rmax,thmax))
+                    meas_ind=[meas_ind,i];%concatenate the indices
+                end
+        end
+            function yes = inview(f,x, rmax, thmax)
+            % Checks for features currently in robot's view
```

Comment:
As long as it is working fine, could you please tell me why it is a bad idea? In opinion, it is a means of grouping modules together that perform similar actions. Here, in this case, the idea of the feature search is to search for the observed features and the "invew" function scans and locates those features. IMHO, it accomplishes a sub part of the whole objective of the feature_search function. 

Infact, In one of my earlier implementations, I had implemented the whole code as one main function with subfunctions. I only separated them as a different files so that they could be shared in future. Having said that, leaving the inview function within the feature search was deliberate.  Also, I love compact codes, hence, I have an inclination towards functions within functions..but thats just me. 

Also, if u read the official documentation of [MATLAB],(http://www.mathworks.com/help/matlab/matlab_prog/local-functions.html), you can see that local functions have precedence over other files and functions. And since, there are no other "invew" functions defined in the scope of fast slam, it should have been straightforward to understand which function it was calling.